lib/cancan/ability.rb

```diff
@@ -252,6 +252,20 @@ def merge(ability)
       self
     end
 
+    def list_can(filter_action = nil)
+      can_do_list = list_abilities[:can]
+      filter_action ? can_do_list[filter_action] : can_do_list
+    end
+
+    def list_cannot(filter_action = nil)
+      cannot_do_list = list_abilities[:cannot]
+      filter_action ? cannot_do_list[filter_action] : cannot_do_list
+    end
+
+    def list_all
```

Comment:
list_all just aliases to a private method. Instead we should just have one public method.